packages/component/src/SendBox/ConnectivityStatus.js

```diff
@@ -18,19 +18,21 @@ class connectConnectivityAlert extends React.Component {
     }
 
     if ( status === 'uninitialized' || ( status === 'connected' && timeoutCompleted === false && connectingSlow === false ) ) {
-      this.timerId = setTimeout( () => {
-        if ( status === 'connected' ) {
-          this.props.dispatch( { type: 'DIRECT_LINE/CONNECT_TIMEOUT_COMPLETE' } );
-        }
+      if ( status === 'connected' ) {
+        this.timerId = this.dispatchConnectivityTimeout();
```

Comment:
won't `this.timerId` be `undefined` in this case because `this.dispatchConnectivityTimeout()` does not return `setTimeout()` it just invokes it

Reply:
I also am having trouble understanding why `this.timerId` is set and then immediately after is set to `null`